code/modules/modular_computers/computers/machinery/modular_computer.dm

```diff
@@ -73,6 +73,11 @@
 		. += "bsod"
 		. += "broken"
 
+/// Eats the "source" arg because update_icon actually expects args now.
+/obj/machinery/modular_computer/proc/relay_icon_update(obj/item/modular_computer/processor/source, updates, updated)
```

Comment:
```suggestion
/obj/machinery/modular_computer/proc/relay_icon_update(datum/source, updates, updated)
```

I'm pretty sure it is not guaranteed that your source will be what you expect it to be in this case, anything else could update the icon.